### ISSUE TITLE
Unable to expect no calls

### ISSUE THREAD

**Issue by imRentable** [zio/zio-mock#27]
Hi, 

I am trying to write a test that verifies that a specific function is not being invoked. Unfortunately, both `zio.mock.Mock#empty` and `zio.mock.Expectation#atMost` make the test succeed although the function in question has been called. Please see the following example.

```
import zio.mock.mockable
import zio.test._
import zio.{Scope, UIO, URIO, ZIO}

trait Foo {
  def doSomething: UIO[Unit]
}

@mockable[Foo]
object FooMock

object MocksDemo extends ZIOSpecDefault {

  def doSomething: URIO[Foo, Unit] = ZIO.serviceWith[Foo](_.doSomething)

  val testCase = test("expect no call to `doSomething`")(doSomething *> assertCompletesZIO)

  override def spec: Spec[TestEnvironment with Scope, Any] = suite("ZIO mocks")(
    testCase.provideLayer(FooMock.empty),
    testCase.provideLayer(FooMock.DoSomething().atMost(0))
  )

}
```

I'd expect that both test cases fail but they do not.

I use ZIO Mock version `1.0.0-RC8` in conjunction with ZIO version `2.0.0`

**Comment by myazinn** [comment]
Hey @imRentable
It looks like the issue is that you are using `ZIO.serviceWith` instead of `ZIO.serviceWithZIO` (it was renamed in ZIO 2). So even though `doSomething` method is invoked, the returned effect is not ran by the runtime, so "effectively" it's not invoked.
I've actually faced this issue with regular mocks from the opposite side: I needed to verify that if we have piece of code like this
```scala
ZIO.serviceWithZIO[Foo](_.doSomething.when(false))
```
then `doSomething` is not invoked. Mockito thought it was invoked, so I had to find other ways to test it. Also note that if zio mocks start counting your example as method invocation, it's super easy to miss a bug when the method returning unit is invoked but not ran, thus leading to a code that passes tests but doesn't actually do anything.
Hope everything above will somehow help you.
